logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/UpdateSelfUserSupportedProtocolsUseCase.kt

```diff
@@ -51,12 +56,23 @@ internal class UpdateSelfUserSupportedProtocolsUseCaseImpl(
     private val userConfigRepository: UserConfigRepository,
     private val featureSupport: FeatureSupport,
     private val currentClientIdProvider: CurrentClientIdProvider,
+    private val mlsClientProvider: MLSClientProvider,
+    private val mlsPublicKeysRepository: MLSPublicKeysRepository,
     private val logger: KaliumLogger
 ) : UpdateSelfUserSupportedProtocolsUseCase {
 
     override suspend operator fun invoke(): Either<CoreFailure, Boolean> {
-        return if (!featureSupport.isMLSSupported) {
-            logger.d("Skip updating supported protocols, since MLS is not supported.")
+        val mlsKey = mlsClientProvider.getMLSClient().flatMap { mlsClient ->
+            val cipherSuite: CipherSuite = CipherSuite.fromTag(mlsClient.getDefaultCipherSuite())
+            mlsPublicKeysRepository.getKeyForCipherSuite(cipherSuite)
+        }
+
+        return if (!featureSupport.isMLSSupported || mlsKey.isLeft()) {
+            logger.d(
+                "Skip updating supported protocols, since MLS is not supported. " +
+                        "Feature flag: ${featureSupport.isMLSSupported} " +
+                        "Has mls key: ${mlsKey.isRight()}"
+            )
```

Comment:
question: not sure what this changes from the code before?
Creating an object of MLS client will create some local files/db for MLS and if there is no need to create it uus better to not init mls untill it is needed

Reply:
+ looking at the Jira ticket i think the app is already doing what it described in the ticket